### PR TITLE
Adds Linked wikis

### PR DESCRIPTION
**Pull request by Royal-lobster**
Closes to https://github.com/EveripediaNetwork/issues/issues/930

# Changes
- made wiki edit details modal more compact and scrollable for additional sections
- refactored files surroundings editors for more discoverability and readability
- adds linked wiki section in create wiki page to link to other wikis

# Screenshots
CleanShot 2022-12-22 at 21 15 00@2x
CleanShot 2022-12-23 at 00 23 01@2x

# Todo
 - [x] Make edit details modal more compact to get space for new sections
 - [x] Brainstorm possible ways to input wiki links
 - [x] Add wiki link input to edit details modal 
 - [x] show case linked wikis in edit details modal
 - [x]  integrate new query to getWikiById 

fixes: https://github.com/EveripediaNetwork/issues/issues/952

### CODE REVIEW

**Review by Simplypheyie96 — APPROVED** [review]
it looks great. 
I assume that there is a limit to the number of linked wikis that can be added right? @Royal-lobster 
Or maybe a scroll bar inside the modal for when the content gets bigger.

**Comment by Royal-lobster** [comment]
> it looks great. I assume that there is a limit to the number of linked wikis that can be added right? @Royal-lobster Or maybe a scroll bar inside the modal for when the content gets bigger.

yes ! the modal is scrollable if the content inside gets longer.

**Review by Adebesin-Cell — APPROVED** [review]
LG 🚀

**Review by Softdev1 — CHANGES_REQUESTED** [review]
I think the prediction is not fetching the required names 
SCR-20230106-np5

**Comment by Royal-lobster** [comment]
> I think the prediction is not fetching and it is not adding the required names SCR-20230106-np5

Thanks for the review ! now it should be fixed.

File: src/components/Editor/EditorModals/EditWikiDetailsModal/LinkedWikisInput.tsx [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+import {
+  AutoComplete,
+  AutoCompleteInput,
+  AutoCompleteItem,
+  AutoCompleteList,
+} from '@choc-ui/chakra-autocomplete'
+import { store } from '@/store/store'
+import { RiArrowRightDownLine, RiCloseLine } from 'react-icons/ri'
+
+const LinkedWikisInput = ({ wiki }: { wiki: Wiki }) => {
```
Comment (kesar) [inline src/components/Editor/EditorModals/EditWikiDetailsModal/LinkedWikisInput.tsx:29]: 245 LoC component, another possible refactoring here 

Not needed to fix for merge 👍🏻

File: src/components/Landing/HeroCard.tsx [hunk 2]
```diff
@@ -56,7 +56,8 @@ export const HeroCard = ({ wiki }: { wiki: Wiki | undefined }) => {
             <chakra.span>{wiki?.title}</chakra.span>
           </LinkOverlay>
           <Text fontSize="xs" color="homeDescriptionColor" my={2}>
-            {wiki && getWikiSummary(wiki, CARD_DETAILS_LENGTH)}
+            {wiki &&
+              getWikiSummary(wiki, CARD_DETAILS_LENGTH as WikiSummarySize)}
```
Comment (kesar) [inline src/components/Landing/HeroCard.tsx:60]: cant you define CARD_DETAILS_LENGTH as WikiSummarySize on intialization? that would make it easier to read here

File: src/components/Landing/TrendingWikis.tsx [hunk 3]
```diff
@@ -87,7 +87,7 @@ const TrendingWikiCard = ({ wiki }: { wiki: Wiki }) => {
               color="homeDescriptionColor"
               my={2}
             >
-              {wiki && getWikiSummary(wiki, 60)}
+              {wiki && getWikiSummary(wiki, 60 as WikiSummarySize)}
```
Comment (kesar) [inline src/components/Landing/TrendingWikis.tsx:90]: doesnt work fine 65 ? so you can use WikiSummarySize.Small ?

Comment (Royal-lobster) [reply src/components/Landing/TrendingWikis.tsx:90]: Oh yes i believe 65 will look fine. i will create a new commit for fix.

File: src/components/Landing/HeroCard.tsx (hunk 2 above)
Comment (Royal-lobster) [reply src/components/Landing/HeroCard.tsx:60]: Thanks ! will add a fix for this.

**Pull request merged**